app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -0,0 +1,116 @@
+module MiqServer::WorkerManagement::Monitor::Kubernetes
+  extend ActiveSupport::Concern
+  attr_accessor :pod_resource_version
+
+  def current_pods
+    @current_pods ||= Concurrent::Hash.new
```

Comment:
We may want Concurrent::Map instead, which has check-and-set functionality (which avoids the `||=` issue)

Reply:
Additionally, since `||=` is not thread safe and can lead to weird behaviour...it's better to just create 1 instance at the start before anything starts, and then always be sure to not replace it.


For example:
```
attr_reader :current_pods

def initialize
  @current_pods = Concurrent::Hash.new
  super
end
```

Then don't ever kill that reference.

Reply:
I was avoiding having this pollute the server instance with it when instantiated.  I guess it's low risk if it's uniquely named and light weight.  I'll try it out. Thanks.